heron/cgi-bin/heron-wsgi.py

```diff
@@ -0,0 +1,397 @@
+#!/usr/bin/env python
+#
+# heron-wsgi - generic RESTful services for Heron
+#
+# This wsgi-application provides Heron services that can only/better be handled
+# within a server. The setup is generic: a parameter 'action' determines
+# which function is to be called. The other parameters are specific to
+# each handler. When using any data conversions and reprojections, the program
+# ogr2ogr from GDAL/OGR (www.gdal.org) is required to be installed on your system.
+
+# Global var: name or path of the GDAL/OGR utility: ogr2ogr
+# In many cases the program is found via the PATH variable,
+# but in some cases, mostly with custom installs, one may change
+# this variable to the full pathname where ogr2ogr resides.
+# For example: OGR2OGR_PROG = '/usr/local/bin/ogr2ogr'.
+# Also note that the GDAL_DATA global var may be required for reprojections.
+OGR2OGR_PROG = 'ogr2ogr'
+
+import cgi
+import base64
+import zipfile
+import subprocess
+import os
+import tempfile
+import sys
+import shutil
+from StringIO import StringIO
+import urllib
+
+
+def print_err(*args):
+    sys.stderr.write(' '.join(map(str,args)) + '\n')
+
+
+def send_error(reason='Unknown'):
+    raise Exception(reason)
```

Comment:
It might be better to handle exceptions where they are raised. But for now I made the minimum changes to get things working under wsgi.